internal/cutil/gocslenslice.go

```diff
@@ -0,0 +1,84 @@
+package cutil
+
+// #include <stdlib.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// GoCSLenSlice is a helper structure that holds Go-allocated slices of
+// C-allocated strings and their respective lengths. Useful for C functions
+// that consume byte buffers with explicit length instead of null-terminated
+// strings. When used as input arguments in C functions, caller must make sure
+// the C code will not hold any pointers to either of the struct's attributes
+// after that C function returns.
+type GoCSLenSlice struct {
```

Comment:
This name strikes me as rather awkward. If I didn't read the full docstring or implementation I do not think I'd know what the type did.  I'd like to change it and some suggestions follow. You can take them, push back, or riff on them:
`CStrSlices`  - closer to the 1st sentence of the doc string
`BufferHolder` - closer to the "intent" of the type I think
`StrGroup` - because it manages the strings as a group

Reply:
I completely agree. I think something with "buffer" would be better. I'll try to think about it.

Reply:
I'm going for `BufferGroup`. Please let me know if you or anyone has some other suggestions.